tests/foreman/api/test_smartproxy.py

```diff
@@ -233,29 +234,66 @@ def test_positive_refresh_features(request, target_sat):
 
 @pytest.mark.skip_if_not_set('fake_capsules')
 @pytest.mark.tier2
-def test_positive_import_puppet_classes(session_puppet_enabled_sat, puppet_proxy_port_range):
-    """Import puppet classes from proxy
+def test_positive_import_puppet_classes(
+    request,
+    session_puppet_enabled_sat,
+    puppet_proxy_port_range,
+    module_puppet_org,
+    module_puppet_loc,
+):
+    """Import puppet classes from proxy for admin and non-admin user
 
     :id: 385efd1b-6146-47bf-babf-0127ce5955ed
 
     :expectedresults: Puppet classes are imported from proxy
 
     :CaseLevel: Integration
 
-    :BZ: 1398695
+    :BZ: 1398695, 2142555
     """
-    with session_puppet_enabled_sat as puppet_sat:
-        new_port = puppet_sat.available_capsule_port
-        with puppet_sat.default_url_on_new_port(9090, new_port) as url:
-            proxy = puppet_sat.api.SmartProxy(url=url).create()
-            result = proxy.import_puppetclasses()
-            assert (
-                "Successfully updated environment and puppetclasses from "
-                "the on-disk puppet installation"
-            ) in result['message'] or "No changes to your environments detected" in result[
-                'message'
+    puppet_sat = session_puppet_enabled_sat
+    # Create role, add permissions and create non-admin user
+    user_login = gen_string('alpha')
+    user_password = gen_string('alpha')
+    role = puppet_sat.api.Role().create()
+    puppet_sat.api_factory.create_role_permissions(
+        role,
+        {
+            'ForemanPuppet::Puppetclass': [
+                'view_puppetclasses',
+                'create_puppetclasses',
+                'import_puppetclasses',
             ]
-        puppet_sat.api.SmartProxy(id=proxy.id).delete()
+        },
+    )
+    user = puppet_sat.api.User(
+        role=[role],
+        admin=False,
+        login=user_login,
+        password=user_password,
+        organization=[module_puppet_org],
+        location=[module_puppet_loc],
+    ).create()
+    request.addfinalizer(user.delete)
+    request.addfinalizer(role.delete)
+
+    new_port = puppet_sat.available_capsule_port
+    with puppet_sat.default_url_on_new_port(9090, new_port) as url:
+        proxy = puppet_sat.api.SmartProxy(url=url).create()
+        result = proxy.import_puppetclasses()
```

Comment:
for non-admin you are passing `enviornment` why this is the case for admin user ?

Reply:
I think, passing the environment to the non-admin user is for adding automation coverage mentioned in BZ[#2142555](https://bugzilla.redhat.com/show_bug.cgi?id=2142555), making API endpoint.

Reply:
updated test to cover both w/ and w/o environment scenarios to import puppetclasses for both admin and non-admin users, thanks!